### PR TITLE
feat: extract data

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: lib/ckb/wallet.rb [hunk 1]
```diff
@@ -131,7 +129,8 @@ def deposit_to_dao(capacity, key: nil)
       i = gather_inputs(
         capacity,
         output.calculate_min_capacity,
-        change_output.calculate_min_capacity
+        change_output.calculate_min_capacity,
+        0
```
Comment (shaojunda) [inline lib/ckb/wallet.rb]: Is it possible to set a default value for the fee?

Comment (classicalliu) [reply lib/ckb/wallet.rb]: `gather_inputs` is an inner method, I think it doesn't matter.

**Comment by ashchan** [comment]
Please note this won't go into `v0.18.0` (correct me if I'm wrong).

**Comment by classicalliu** [comment]
> Please note this won't go into `v0.18.0` (correct me if I'm wrong). If we merge before Aug 10th please make sure 0.18.0 doesn't contain this change.

Yes, it's for v0.19-pre1, I think we can checkout v0.18 from v0.17, and develop for new features if anyone need.

**Comment by ashchan** [comment]
> > Please note this won't go into `v0.18.0` (correct me if I'm wrong). If we merge before Aug 10th please make sure 0.18.0 doesn't contain this change.
> 
> Yes, it's for v0.19-pre1, I think we can checkout v0.18 from v0.17, and develop for new features if anyone need.

OK. We can also hold this until next Monday (after `v0.18.0` release) then merge it.

**Pull request merged**